### PR TITLE
Add filter to clearcommands

### PR DESCRIPTION
**Pull request by marceloneil**
TODO:
- [x] Add a filter to the CLI `clear-commands`
- [x] Update api docs
- [x] Add support for conversation type advertisements

The API is similar:
Clear all advertisements
```
{"method": "clearcommands"}
```
Clear only public advertisements
```
{"method": "clearcommands", "params": {"options": {"filter": {"type": "public"}}}}
```
Clear advertisements for teamconv/teammembers:
```
{"method": "clearcommands", "params": {"options": {"filter": {"type": "teamconvs", "team_name": "someteam"}}}}
```
Clear advertisements for conv:
```
{"method": "clearcommands", "params": {"options": {"filter": {"type": "conv", "conv_id": "someconvid"}}}}
```

The CLI looks like:
```
keybase chat clear-commands
keybase chat clear-commands --type public
keybase chat clear-commands --type teamconvs --team-name treehouse
keybase chat clear-commands --type conv --team-name treehouse --channel random
```

### CODE REVIEW

**Comment by joshblum** [comment]
can you base off of https://github.com/keybase/client/pull/23872, going to merge after CI

File: go/chat/bots/commands.go [hunk 1]
```diff
@@ -154,46 +154,48 @@ func (b *CachingBotCommandManager) deriveMembersType(ctx context.Context, name s
 	}
 }
 
-func (b *CachingBotCommandManager) createConv(ctx context.Context, param chat1.AdvertiseCommandsParam) (res chat1.ConversationLocal, err error) {
+func (b *CachingBotCommandManager) createConv(
+	ctx context.Context, typ chat1.BotCommandsAdvertisementTyp, teamName *string, convID *chat1.ConversationID,
```
Comment (joshblum) [inline go/chat/bots/commands.go:158]: i think it's nice to wrap long function signatures but i'm not a fan of newlines between everything here in general

File: go/chat/bots/commands.go [hunk 2]
```diff
@@ -273,9 +275,32 @@ func (b *CachingBotCommandManager) Advertise(ctx context.Context, alias *string,
...
 }
 
-func (b *CachingBotCommandManager) Clear(ctx context.Context) (err error) {
+func (b *CachingBotCommandManager) Clear(ctx context.Context, filter *chat1.ClearBotCommandsFilter) (err error) {
 	defer b.Trace(ctx, &err, "Clear")()
-	if _, err := b.ri().ClearBotCommands(ctx); err != nil {
+	var remote *chat1.RemoteClearBotCommandsFilter
+	if filter != nil {
+		remote = new(chat1.RemoteClearBotCommandsFilter)
+		conv, err := b.createConv(ctx, filter.Typ, filter.TeamName, filter.ConvID)
```
Comment (joshblum) [inline go/chat/bots/commands.go:283]: we only need to do this in the tlfid cases

File: go/protocol/chat1/extras.go [hunk 3]
```diff
@@ -3163,6 +3163,38 @@ func (p AdvertiseCommandsParam) ToRemote(cmdConvID ConversationID, tlfID *TLFID,
 	}
 }
 
+func (p ClearBotCommandsFilter) ToRemote(tlfID *TLFID, convID *ConversationID) (res RemoteClearBotCommandsFilter, err error) {
+	switch p.Typ {
+	case BotCommandsAdvertisementTyp_PUBLIC:
+		return NewRemoteClearBotCommandsFilterWithPublic(RemoteClearBotCommandsFilterPublic{}), nil
```
Comment (joshblum) [inline go/protocol/chat1/extras.go:3169]: can you sanity check that the other args are nil?

File: go/protocol/chat1/extras.go [hunk 4]
```diff
@@ -3163,6 +3163,38 @@ func (p AdvertiseCommandsParam) ToRemote(cmdConvID ConversationID, tlfID *TLFID,
...
+	case BotCommandsAdvertisementTyp_TLFID_CONVS:
+		if tlfID == nil {
+			return res, errors.New("no TLFID specified")
+		}
```
Comment (joshblum) [inline go/protocol/chat1/extras.go:3173]: error out if conv id is present

File: go/protocol/chat1/extras.go [hunk 5]
```diff
@@ -3163,6 +3163,38 @@ func (p AdvertiseCommandsParam) ToRemote(cmdConvID ConversationID, tlfID *TLFID,
...
+		return NewRemoteClearBotCommandsFilterWithTlfidConvs(RemoteClearBotCommandsFilterTLFID{
+			TlfID: *tlfID,
+		}), nil
+	case BotCommandsAdvertisementTyp_TLFID_MEMBERS:
+		if tlfID == nil {
+			return res, errors.New("no TLFID specified")
+		}
```
Comment (joshblum) [inline go/protocol/chat1/extras.go:3180]: error out if conv id is present

**Comment by marceloneil** [comment]
@joshblum do you think the CLI should error if there are any extraneous CLI flags (ie. a `--team-name` when using `--type public`)

**Comment by joshblum** [comment]
@marceloneil i dont think it's necessary, the service will reject the request if so. Can you wait to merge until gregor is deployed?

**Comment by marceloneil** [comment]
> @marceloneil i dont think it's necessary, the service will reject the request if so. Can you wait to merge until gregor is deployed?

The way it's set up right now, the service won't reject it (the team-name will simply be dropped). Is it still okay? Example: 
```
kb chat clear-commands --type public --team-name test
▶ WARNING Running in devel mode
Cleared bot commands successfully.
```

Re gregor: yes!

**Comment by joshblum** [comment]
we can sanitize here then so it's not confusing to users

**Pull request merged**